compiler/rustc_hir/src/definitions.rs

```diff
@@ -450,8 +474,10 @@ impl DefPathData {
             ImplTrait => sym::double_braced_opaque,
         }
     }
+}
 
-    pub fn to_string(&self) -> String {
-        self.as_symbol().to_string()
+impl fmt::Display for DefPathData {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(&self.as_legacy_symbol().as_str())
```

Comment:
Wait, how is this useful for a `fmt::Display`? It completely erases the disambiguator.